### ISSUE TITLE
Game Object Explorer Error

### ISSUE THREAD

**Issue by Viscerus** [ShinyHobo/BG3-Modders-Multitool#154]
I followed the ReadMe file and unpacked necessary files, I am on version (v0.9.12). My latest log is from two days ago so I unsure if it would help or not, but I can send it if necessary.
I unpacked, decompressed, then indexed and the console outputs this:

```
[08/29/2023 17:04:58] Opening GameObject Explorer. This could take a while.
[08/29/2023 17:05:00] Translations loaded...
[08/29/2023 17:05:00] Loading CharacterVisualBanks from cache...
[08/29/2023 17:05:00] Loading VisualBanks from cache...
[08/29/2023 17:05:00] Loading BodySetVisuals from cache...
[08/29/2023 17:05:00] Loading MaterialBanks from cache...
[08/29/2023 17:05:00] Loading TextureBanks from cache...
[08/29/2023 17:05:00] Reading GameObjects from root templates; this will take a while...
[08/29/2023 17:05:58] GameObjects loaded...
[08/29/2023 17:05:58] GameObjects failed to load. 59.8367957 seconds passed.
[08/29/2023 17:05:58] One or more errors occurred.
[08/29/2023 17:05:58] Object reference not set to an instance of an object.
[08/29/2023 17:05:58]    at bg3_modders_multitool.Models.StatStructures.StatStructure.InheritProperties(String line, List`1 statStructures)
   at bg3_modders_multitool.Services.RootTemplateHelper.ReadData(String pak)
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>d__51.MoveNext()
```

**Comment by ShinyHobo** [comment]
I would suggest re-unpacking, re-decompressing, and re-indexing, starting from a clean slate. I can't tell the specific cause of the issue from this report, but will add more error logging to help narrow it down in the future.

**Comment by dovahkiin-tim** [comment]
Hello I am also getting this error and I have done several clean reinstalls 

[09/02/2023 00:47:36] No GameObject cache to clear.
[09/02/2023 00:54:53] Unpacking processes starting. This could take a while; please wait for all console processes to close on their own.
[09/02/2023 01:15:15] All unpacking processes finished.
[09/02/2023 01:15:15] Unpacking complete!
[09/02/2023 01:15:59] Retrieving file list for decompression.
[09/02/2023 01:16:29] Retrived file list. Starting decompression; this could take awhile.
[09/02/2023 01:29:32] Decompression completed in 00:13:02.80.
[09/02/2023 01:45:53] Retrieving file list.
[09/02/2023 01:46:27] File list retrieved.
[09/02/2023 01:46:27] Indexing in progress...
[09/02/2023 02:22:03] Finalizing index...
[09/02/2023 02:22:53] Indexing process finished in 00:36:26.
[09/02/2023 02:23:28] Opening GameObject Explorer. This could take a while.
[09/02/2023 02:23:29] Translations loaded...
[09/02/2023 02:23:29] Loading bank files...
[09/02/2023 02:23:58] VisualBanks found...
[09/02/2023 02:24:26] MaterialBanks found...
[09/02/2023 02:24:54] TextureBanks found...
[09/02/2023 02:24:54] Sorting bank files...
[09/02/2023 02:47:03] Caching CharacterVisualBanks...
[09/02/2023 02:47:03] Caching VisualBanks...
[09/02/2023 02:47:03] Caching BodySetVisuals...
[09/02/2023 02:47:03] Caching MaterialBanks...
[09/02/2023 02:47:03] Caching TextureBanks...
[09/02/2023 02:47:03] Reading GameObjects from root templates; this will take a while...
[09/02/2023 02:48:54] Patch1\Mods\Benchmarking\Levels\Benchmark_Ext_Avernus\Splines\_merged.lsx appears to be corrupt or missing. Delete file and Decompress, then try again. Skipping file...
[09/02/2023 02:49:03] GameObjects loaded...
[09/02/2023 02:49:05] Sorting GameObjects...
[09/02/2023 02:49:08] GameObjects failed to load. 1539.8244268 seconds passed.
[09/02/2023 02:49:08] One or more errors occurred.
[09/02/2023 02:49:08] Specified argument was out of the range of valid values.
Parameter name: degreeOfParallelism
[09/02/2023 02:49:08]    at System.Linq.ParallelEnumerable.WithDegreeOfParallelism[TSource](ParallelQuery`1 source, Int32 degreeOfParallelism)
   at bg3_modders_multitool.Services.RootTemplateHelper.SortRootTemplate()
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>d__51.MoveNext()
[09/02/2023 08:55:30] Opening GameObject Explorer. This could take a while.
[09/02/2023 08:55:32] Translations loaded...
[09/02/2023 08:55:32] Loading CharacterVisualBanks from cache...
[09/02/2023 08:55:32] Loading VisualBanks from cache...
[09/02/2023 08:55:32] Loading BodySetVisuals from cache...
[09/02/2023 08:55:32] Loading MaterialBanks from cache...
[09/02/2023 08:55:32] Loading TextureBanks from cache...
[09/02/2023 08:55:32] Reading GameObjects from root templates; this will take a while...
[09/02/2023 08:57:23] Patch1\Mods\Benchmarking\Levels\Benchmark_Ext_Avernus\Splines\_merged.lsx appears to be corrupt or missing. Delete file and Decompress, then try again. Skipping file...
[09/02/2023 08:57:31] GameObjects loaded...
[09/02/2023 08:57:32] Sorting GameObjects...
[09/02/2023 08:57:35] GameObjects failed to load. 124.5339338 seconds passed.
[09/02/2023 08:57:35] One or more errors occurred.
[09/02/2023 08:57:35] Specified argument was out of the range of valid values.
Parameter name: degreeOfParallelism
[09/02/2023 08:57:35]    at System.Linq.ParallelEnumerable.WithDegreeOfParallelism[TSource](ParallelQuery`1 source, Int32 degreeOfParallelism)
   at bg3_modders_multitool.Services.RootTemplateHelper.SortRootTemplate()
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>d__51.MoveNext()

**Comment by ShinyHobo** [comment]
@dovahkiin-tim It looks like your cpu isn't recent enough to use this application. Go into the configuration menu and click Unlock Threads.

**Comment by dovahkiin-tim** [comment]
> source, Int32 degreeOfParallelism) at bg3_modders_multitool.Services.RootTemplateHelper.SortRootTemplate() at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0() at

So I have a AMD Ryzen 5 5600X 6-Core Processor 4.20 GHz I had the the unlock threads option on from the getgo so I tried to do it with them off but I still get the same issue 

09/02/2023 12:51:15] Opening GameObject Explorer. This could take a while.
[09/02/2023 12:51:17] Translations loaded...
[09/02/2023 12:51:17] Loading CharacterVisualBanks from cache...
[09/02/2023 12:51:17] Loading VisualBanks from cache...
[09/02/2023 12:51:17] Loading BodySetVisuals from cache...
[09/02/2023 12:51:17] Loading MaterialBanks from cache...
[09/02/2023 12:51:17] Loading TextureBanks from cache...
[09/02/2023 12:51:17] Reading GameObjects from root templates; this will take a while...
[09/02/2023 12:53:08] Patch1\Mods\Benchmarking\Levels\Benchmark_Ext_Avernus\Splines\_merged.lsx appears to be corrupt or missing. Delete file and Decompress, then try again. Skipping file...
[09/02/2023 12:53:21] GameObjects loaded...
[09/02/2023 12:53:22] Sorting GameObjects...
[09/02/2023 12:53:24] GameObjects failed to load. 128.8532524 seconds passed.
[09/02/2023 12:53:24] One or more errors occurred.
[09/02/2023 12:53:24] Specified argument was out of the range of valid values.
Parameter name: degreeOfParallelism
[09/02/2023 12:53:24]    at System.Linq.ParallelEnumerable.WithDegreeOfParallelism[TSource](ParallelQuery`1 source, Int32 degreeOfParallelism)
   at bg3_modders_multitool.Services.RootTemplateHelper.SortRootTemplate()
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>d__51.MoveNext()

**Comment by dovahkiin-tim** [comment]
@ShinyHobo  I disabled Unlock threads and then I deleted the folder mentions above in the error log. Cleared the cache, unpacked and decompressed the deleted folder followed by a new index and then it finally let me view the objects.  Unchecking unlock threads made the process run significantly faster.

**Comment by ShinyHobo** [comment]
I'm glad you got it working. The unlock theads option disables the max concurrent threads a loop can use (setting to -1, or infinite); the tooltip mentions you might not get performance benefits when using it. Perhaps your cpu is incompatible with the with setting.

**Comment by ShinyHobo** [comment]
@Viscerus Did you get this sorted?

**Comment by ShinyHobo** [comment]
@dovahkiin-tim this issue with Unlock Threads will be fixed in 0.10.6. I found the root cause.
